Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -951,15 +937,17 @@ namespace OpenInfraPlatform {
 						else
 							polygon->addFace(v0, v1, v2);
 					}
+					return polygon;
 				}
 
 				/*! \brief Converts \c IfcCartesianPoint to a 2D vector.
 				\param	faceVertices2D to be triangulated using carve
 				\param	faceVertices3D to be added to the triangulation
 				*/
 
-				void convertIfcCartesianPoint2DVector(const std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D,
-					std::vector<std::vector<carve::geom::vector<3>>>& loop2D)  const throw(...) {
+				std::vector<std::vector<carve::geom::vector<3>>> convertIfcCartesianPoint2DVector(const std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D)  const throw(...)
+				{
+					std::vector<std::vector<carve::geom::vector<3>>> loop2D = std::vector<std::vector<carve::geom::vector<3>>>();
 					const double lengthFactor = UnitConvert()->getLengthInMeterFactor();
 					const uint32_t numPointsY = points2D.size();
 					loop2D.resize(numPointsY);

```

Comment:
```suggestion
					loop2D.reserve(numPointsY);
```
Resize in combination with push_back is wrong.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/a7a0a904dae38eedb4769899d6ab204946ee43f8